packages/@aws-cdk/cicd/README.md

```diff
@@ -0,0 +1,112 @@
+* `ServiceStackA` and `ServiceStackB` are service infrastructure stacks, and need to be deployed in this order
+
+```
+  ┏━━━━━━━━━━━━━━━━┓  ┏━━━━━━━━━━━━━━━━┓  ┏━━━━━━━━━━━━━━━━┓  ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+  ┃     Source     ┃  ┃     Build      ┃  ┃  Self-Update   ┃  ┃            Deploy             ┃
+  ┃                ┃  ┃                ┃  ┃                ┃  ┃                               ┃
+  ┃ ┌────────────┐ ┃  ┃ ┌────────────┐ ┃  ┃ ┌────────────┐ ┃  ┃ ┌────────────┐ ┌────────────┐ ┃
+  ┃ │   GitHub   ┣━╋━━╋━▶ CodeBuild  ┣━╋━━╋━▶Deploy Stack┣━╋━━╋━▶Deploy Stack┣━▶Deploy Stack│ ┃
+  ┃ └────────────┘ ┃  ┃ └────────────┘ ┃  ┃ └────────────┘ ┃  ┃ └────────────┘ └────────────┘ ┃
+  ┗━━━━━━━━━━━━━━━━┛  ┗━━━━━━━━━━━━━━━━┛  ┗━━━━━━━━━━━━━━━━┛  ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+```
+
+#### `index.ts`
+```ts
+import codebuild = require('@aws-cdk/aws-codebuild');
+import codepipeline = require('@aws-cdk/aws-codepipeline');
+import cdk = require('@aws-cdk/cdk');
+import cicd = require('@aws-cdk/cicd');
+
+const app = new cdk.App();
+
+// We define a stack that contains the CodePipeline
+const pipelineStack = new cdk.Stack(app, 'CodePipelineStack');
+const pipeline = new codepipeline.Pipeline(pipelineStack, 'CodePipeline', {
+  restartExecutionOnUpdate: true,
+  /* ... */
+});
+// Configure the CodePipeline source - where your CDK App's source code is hosted
+const source = new codepipeline.GitHubSourceAction(pipelineStack, 'GitHub', {
+  stage: pipeline.addStage('source'),
+  /* ... */
+});
+const project = new codebuild.Project(pipelineStack, 'CodeBuild', {
+  artifacts: new codebuild.CodePipelineArtifacts(),
+  source: new codebuild.CodePipelineSource(),
+  /* ... */
+});
+const synthesizedApp = project.outputArtifact;
+
+// Optionally, self-update the pipeline stack
+const selfUpdateStage = pipeline.addStage('SelfUpdate');
+new cicd.PipelineDeployStackAction(pipelineStack, 'SelfUpdatePipeline', {
+  stage: selfUpdateStage,
+  stack: pipelineStack,
+  inputArtifact: synthesizedApp,
+});
+
+// Now add our service stacks
+const deployStage = pipeline.addStage('Deploy');
+const serviceStackA = new MyServiceStackA(app, 'ServiceStackA', { /* ... */ });
+const serviceStackB = new MyServiceStackB(app, 'ServiceStackB', { /* ... */ });
+// Add actions to deploy the stacks in the deploy stage:
+new cicd.PipelineDeployStackAction(pipelineStack, 'DeployServiceStack', {
+  stage: deployStage,
+  stack: serviceStackA,
+  inputArtifact: synthesizedApp,
+  createChangeSetRunOrder: 1,
+  executeChangeSetRunOrder: 2,
+});
+new cicd.PipelineDeployStackAction(pipelineStack, 'DeployServiceStack', {
+  stage: deployStage,
+  stack: serviceStackB,
+  inputArtifact: synthesizedApp,
+  createChangeSetRunOrder: 998,
+  executeChangeSetRunOrder: 999,
+});
+```
+
+#### `buildspec.yml`
+In order for the *CodeBuild* step to produce an input that meets the expectations of the `PipelineDeployStackAction`
```

Comment:
I would first describe the general requirement and then dive into the example. Something like: "PipelineDeployStackAction expects that it's input artifact will contain the output from 'cdk synth -o <out>'. For example, the following buildspec...."